test/TestApps/ToastNotificationsTestApp/main.cpp

```diff
@@ -435,6 +443,171 @@ std::string unitTestNameFromLaunchArguments(const winrt::ILaunchActivatedEventAr
+
+    auto progressResultOperation = winrt::ToastNotificationManager::Default().UpdateToastProgressDataAsync(progressData, L"Tag", L"Group");
+    auto progressResult = ProgressResultOperationHelper(progressResultOperation);
+
+    return progressResult == winrt::ToastProgressResult::Succeeded;
+}
+
+bool VerifyUpdateToastProgressDataUsingValidTagAndEmptyGroup()
+{
+    // Registration happens in main()
+    PostToastHelper(L"PTag", L"");
+
+    winrt::ToastProgressData progressData = GetToastProgressData(L"PStatus", L"PTitle", 0.10, L"10%");
+
+    auto progressResultOperation = winrt::ToastNotificationManager::Default().UpdateToastProgressDataAsync(progressData, L"PTag");
+    auto progressResult = ProgressResultOperationHelper(progressResultOperation);
+
+    return progressResult == winrt::ToastProgressResult::Succeeded;
+}
+
+bool VerifyUpdateToastProgressDataUsingValidTagAndEmptyGroup_Unpackaged()
+{
+    winrt::ToastAssets assets(L"ToastNotificationApp", winrt::Uri{ LR"(C:\Windows\System32\WindowsSecurityIcon.png)" });
+    auto activationInfo = winrt::ToastActivationInfo::CreateFromToastAssets(assets);
+
+    winrt::ToastNotificationManager::Default().RegisterActivator(activationInfo);
+
+    auto scope_exit = wil::scope_exit(
+        [&] {
+            winrt::ToastNotificationManager::Default().UnregisterActivator();
+        });
+
+    PostToastHelper(L"Tag", L"");
+
+    winrt::ToastProgressData progressData = GetToastProgressData(L"SomeStatus", L"SomeTitle", 0.14, L"14%");
+
+    auto progressResultOperation = winrt::ToastNotificationManager::Default().UpdateToastProgressDataAsync(progressData, L"Tag");
+    auto progressResult = ProgressResultOperationHelper(progressResultOperation);
+
+    return progressResult == winrt::ToastProgressResult::Succeeded;
+}
+
+bool VerifyFailedUpdateNotificationDataWithNonExistentTagAndGroup()
+{
+    // Registration happens in main()
+    PostToastHelper(L"PTag", L"PGroup");
+
+    winrt::ToastProgressData progressData = GetToastProgressData(L"PStatus", L"PTitle", 0.10, L"10%");
+
+    auto progressResultOperation = winrt::ToastNotificationManager::Default().UpdateToastProgressDataAsync(progressData, L"NonExistentTag", L"NonExistentGroup");
+    auto progressResult = ProgressResultOperationHelper(progressResultOperation);
+
+    return progressResult == winrt::ToastProgressResult::NotificationNotFound;
+}
+
+bool VerifyFailedUpdateNotificationDataWithNonExistentTagAndGroup_Unpackaged()
+{
+    winrt::ToastAssets assets(L"ToastNotificationApp", winrt::Uri{ LR"(C:\Windows\System32\WindowsSecurityIcon.png)" });
+    auto activationInfo = winrt::ToastActivationInfo::CreateFromToastAssets(assets);
+
+    winrt::ToastNotificationManager::Default().RegisterActivator(activationInfo);
+
+    auto scope_exit = wil::scope_exit(
+        [&] {
+            winrt::ToastNotificationManager::Default().UnregisterActivator();
+        });
+
+    PostToastHelper(L"Tag", L"Group");
```

Comment:
Any tests for empty tag but valid group?

Reply:
That test should return Not_Found as part of enum because there is no toast to replace.